[BUG] SAMM-CLI v2.3.1 (also v2.2.2, v2.2.2) BAMM 2.0.0 models referencing other BAMM 2.0.0 models are not properly resolved

**Describe the bug**
Exception thrown for
```
java -jar ~/Downloads/samm-cli-2.3.1.jar  aspect io.catenax.refurbishing_certificate/1.0.0/RefurbishingCertificate.ttl validate
```

when a BAMM 2.0.0 model references another BAMM 2.0.0 model.

stack trace
```
org.eclipse.esmf.aspectmodel.resolver.ModelResolutionException: Could not resolve [urn:bamm:io.openmanufacturing:meta-model:2.0.0#Aspect, urn:bamm:io.openmanufacturing:meta-model:2.0.0#properties, urn:bamm:io.openmanufacturing:meta-model:2.0.0#events, urn:bamm:io.openmanufacturing:meta-model:2.0.0#preferredName, urn:bamm:io.openmanufacturing:meta-model:2.0.0#description, urn:bamm:io.openmanufacturing:meta-model:2.0.0#operations, urn:bamm:io.openmanufacturing:meta-model:2.0.0#characteristic, urn:bamm:io.catenax.shared.recycling_strategy_certificate:1.0.0#RecyclingStrategyCertificateCharacteristic, urn:bamm:io.openmanufacturing:meta-model:2.0.0#Property]
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.resolveAspectModel(AspectModelResolver.java:176)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.resolveAspectModel(AspectModelResolver.java:146)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.lambda$resolveAspectModel$1(AspectModelResolver.java:126)
	at io.vavr.control.Try.flatMapTry(Try.java:490)
	at io.vavr.control.Try.flatMap(Try.java:472)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.resolveAspectModel(AspectModelResolver.java:126)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.lambda$loadAndResolveModel$8(AspectModelResolver.java:363)
	at io.vavr.control.Try.flatMapTry(Try.java:490)
	at io.vavr.control.Try.flatMap(Try.java:472)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.loadAndResolveModel(AspectModelResolver.java:361)
	at org.eclipse.esmf.AbstractCommand.loadAndResolveModel(AbstractCommand.java:53)
	at org.eclipse.esmf.aspect.AspectValidateCommand.run(AspectValidateCommand.java:60)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2026)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at org.eclipse.esmf.LoggingMixin.executionStrategy(LoggingMixin.java:41)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at org.eclipse.esmf.SammCli.main(SammCli.java:106)
Caused by: java.io.FileNotFoundException: No model file containing urn:samm:io.catenax.shared.recycling_strategy_certificate:1.0.0#RecyclingStrategyCertificateCharacteristic could be found in directory: /Users/johannes/Documents/GitHub/sldt-semantic-models/io.catenax.shared.recycling_strategy_certificate/1.0.0
	at org.eclipse.esmf.aspectmodel.resolver.FileSystemStrategy.apply(FileSystemStrategy.java:93)
	at org.eclipse.esmf.aspectmodel.resolver.FileSystemStrategy.apply(FileSystemStrategy.java:33)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.getModelForUrn(AspectModelResolver.java:311)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.resolve(AspectModelResolver.java:243)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.resolveAspectModel(AspectModelResolver.java:171)
	... 21 more
```

**Where**
Model resolution in SAMM SDK (as error indicates)
it tries to find a urn:samm but should be urn:bam
```
urn:samm:io.catenax.shared.recycling_strategy_certificate:1.0.0#RecyclingStrategyCertificateCharacteristic could be found in directory: /Users/johannes/Documents/GitHub/sldt-semantic-models/io.catenax.shared.recycling_strategy_certificate/1.0.0

```

Can be reproduced with model
https://github.com/eclipse-tractusx/sldt-semantic-models/blob/main/io.catenax.refurbishing_certificate/1.0.0/RefurbishingCertificate.ttl

(repo https://github.com/eclipse-tractusx/sldt-semantic-models.git) needs to be checked out to allow to find the referenced model.

**Screenshots**
If applicable, add screenshots to help explain your problem.


**Additional context**
Add any other context about the problem here.

@michelu89 @RaMisess it seems the bug is still present in v2.3.2. At least I get the same error as with 2.3.2
```
org.eclipse.esmf.aspectmodel.resolver.ModelResolutionException: Could not resolve [urn:bamm:io.openmanufacturing:meta-model:2.0.0#Aspect, urn:bamm:io.openmanufacturing:meta-model:2.0.0#properties, urn:bamm:io.openmanufacturing:meta-model:2.0.0#events, urn:bamm:io.openmanufacturing:meta-model:2.0.0#preferredName, urn:bamm:io.openmanufacturing:meta-model:2.0.0#description, urn:bamm:io.openmanufacturing:meta-model:2.0.0#operations, urn:bamm:io.openmanufacturing:meta-model:2.0.0#characteristic, urn:bamm:io.catenax.shared.recycling_strategy_certificate:1.0.0#RecyclingStrategyCertificateCharacteristic, urn:bamm:io.openmanufacturing:meta-model:2.0.0#Property]
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.resolveAspectModel(AspectModelResolver.java:188)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.resolveAspectModel(AspectModelResolver.java:156)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.lambda$resolveAspectModel$1(AspectModelResolver.java:133)
	at io.vavr.control.Try.flatMapTry(Try.java:490)
	at io.vavr.control.Try.flatMap(Try.java:472)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.resolveAspectModel(AspectModelResolver.java:133)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.lambda$loadAndResolveModelFromDir$33570ba8$1(AspectModelResolver.java:414)
	at io.vavr.control.Try$WithResources1.lambda$of$97fea75d$1(Try.java:1392)
	at io.vavr.control.Try.of(Try.java:75)
	at io.vavr.control.Try$WithResources1.of(Try.java:1390)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.lambda$loadAndResolveModelFromDir$53c64c99$1(AspectModelResolver.java:414)
	at io.vavr.control.Try.flatMapTry(Try.java:490)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.loadAndResolveModelFromDir(AspectModelResolver.java:412)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.lambda$loadAndResolveModel$9(AspectModelResolver.java:385)
	at io.vavr.control.Try.orElse(Try.java:726)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.loadAndResolveModel(AspectModelResolver.java:385)
	at org.eclipse.esmf.AbstractCommand.loadAndResolveModel(AbstractCommand.java:53)
	at org.eclipse.esmf.aspect.AspectValidateCommand.run(AspectValidateCommand.java:60)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2026)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at org.eclipse.esmf.LoggingMixin.executionStrategy(LoggingMixin.java:41)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at org.eclipse.esmf.SammCli.main(SammCli.java:106)
Caused by: java.io.FileNotFoundException: No model file containing urn:samm:io.catenax.shared.recycling_strategy_certificate:1.0.0#RecyclingStrategyCertificateCharacteristic could be found in directory: /Users/johannes/Documents/GitHub/sldt-semantic-models/io.catenax.refurbishing_certificate/1.0.0/io.catenax.shared.recycling_strategy_certificate/1.0.0. AND No model file containing urn:samm:io.catenax.shared.recycling_strategy_certificate:1.0.0#RecyclingStrategyCertificateCharacteristic could be found in directory: /Users/johannes/Documents/GitHub/sldt-semantic-models/io.catenax.refurbishing_certificate/1.0.0/io.catenax.shared.recycling_strategy_certificate/1.0.0
	at org.eclipse.esmf.aspectmodel.resolver.FileSystemStrategy$DefaultNamespace.lambda$apply$0(FileSystemStrategy.java:137)
	at io.vavr.API$Match$Case0.apply(API.java:5135)
	at io.vavr.API$Match.option(API.java:5105)
	at io.vavr.control.Try.mapFailure(Try.java:602)
	at org.eclipse.esmf.aspectmodel.resolver.FileSystemStrategy$DefaultNamespace.lambda$apply$1(FileSystemStrategy.java:136)
	at io.vavr.control.Try.recoverWith(Try.java:858)
	at org.eclipse.esmf.aspectmodel.resolver.FileSystemStrategy$DefaultNamespace.apply(FileSystemStrategy.java:134)
	at org.eclipse.esmf.aspectmodel.resolver.FileSystemStrategy$DefaultNamespace.apply(FileSystemStrategy.java:116)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.getModelForUrn(AspectModelResolver.java:332)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.resolve(AspectModelResolver.java:262)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.resolveAspectModel(AspectModelResolver.java:182)
	... 27 more
Johanness-MacBook-Pro:sldt-semantic-models johannes$ java -jar ~/Downloads/samm-cli-2.3.2.jar  aspect io.catenax.refurbishing_certificate/1.0.0/RefurbishingCertificate.ttl validate
[To redirect Truffle log output to a file use one of the following options:
* '--log.file=<path>' if the option is passed using a guest language launcher.
* '-Dpolyglot.log.file=<path>' if the option is passed using the host Java launcher.
* Configure logging using the polyglot embedding API.]
[engine] WARNING: The polyglot context is using an implementation that does not support runtime compilation.
The guest application code will therefore be executed in interpreted mode only.
Execution only in interpreted mode will strongly impact the guest application performance.
For more information on using GraalVM see https://www.graalvm.org/java/quickstart/.
To disable this warning the '--engine.WarnInterpreterOnly=false' option or use the '-Dpolyglot.engine.WarnInterpreterOnly=false' system property.
org.eclipse.esmf.aspectmodel.resolver.ModelResolutionException: Could not resolve [urn:bamm:io.openmanufacturing:meta-model:2.0.0#Aspect, urn:bamm:io.openmanufacturing:meta-model:2.0.0#properties, urn:bamm:io.openmanufacturing:meta-model:2.0.0#events, urn:bamm:io.openmanufacturing:meta-model:2.0.0#preferredName, urn:bamm:io.openmanufacturing:meta-model:2.0.0#description, urn:bamm:io.openmanufacturing:meta-model:2.0.0#operations, urn:bamm:io.openmanufacturing:meta-model:2.0.0#characteristic, urn:bamm:io.catenax.shared.recycling_strategy_certificate:1.0.0#RecyclingStrategyCertificateCharacteristic, urn:bamm:io.openmanufacturing:meta-model:2.0.0#Property]
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.resolveAspectModel(AspectModelResolver.java:188)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.resolveAspectModel(AspectModelResolver.java:156)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.lambda$resolveAspectModel$1(AspectModelResolver.java:133)
	at io.vavr.control.Try.flatMapTry(Try.java:490)
	at io.vavr.control.Try.flatMap(Try.java:472)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.resolveAspectModel(AspectModelResolver.java:133)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.lambda$loadAndResolveModelFromDir$33570ba8$1(AspectModelResolver.java:414)
	at io.vavr.control.Try$WithResources1.lambda$of$97fea75d$1(Try.java:1392)
	at io.vavr.control.Try.of(Try.java:75)
	at io.vavr.control.Try$WithResources1.of(Try.java:1390)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.lambda$loadAndResolveModelFromDir$53c64c99$1(AspectModelResolver.java:414)
	at io.vavr.control.Try.flatMapTry(Try.java:490)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.loadAndResolveModelFromDir(AspectModelResolver.java:412)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.lambda$loadAndResolveModel$9(AspectModelResolver.java:385)
	at io.vavr.control.Try.orElse(Try.java:726)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.loadAndResolveModel(AspectModelResolver.java:385)
	at org.eclipse.esmf.AbstractCommand.loadAndResolveModel(AbstractCommand.java:53)
	at org.eclipse.esmf.aspect.AspectValidateCommand.run(AspectValidateCommand.java:60)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2026)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at org.eclipse.esmf.LoggingMixin.executionStrategy(LoggingMixin.java:41)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at org.eclipse.esmf.SammCli.main(SammCli.java:106)
Caused by: java.io.FileNotFoundException: No model file containing urn:samm:io.catenax.shared.recycling_strategy_certificate:1.0.0#RecyclingStrategyCertificateCharacteristic could be found in directory: /Users/johannes/Documents/GitHub/sldt-semantic-models/io.catenax.refurbishing_certificate/1.0.0/io.catenax.shared.recycling_strategy_certificate/1.0.0. AND No model file containing urn:samm:io.catenax.shared.recycling_strategy_certificate:1.0.0#RecyclingStrategyCertificateCharacteristic could be found in directory: /Users/johannes/Documents/GitHub/sldt-semantic-models/io.catenax.refurbishing_certificate/1.0.0/io.catenax.shared.recycling_strategy_certificate/1.0.0
	at org.eclipse.esmf.aspectmodel.resolver.FileSystemStrategy$DefaultNamespace.lambda$apply$0(FileSystemStrategy.java:137)
	at io.vavr.API$Match$Case0.apply(API.java:5135)
	at io.vavr.API$Match.option(API.java:5105)
	at io.vavr.control.Try.mapFailure(Try.java:602)
	at org.eclipse.esmf.aspectmodel.resolver.FileSystemStrategy$DefaultNamespace.lambda$apply$1(FileSystemStrategy.java:136)
	at io.vavr.control.Try.recoverWith(Try.java:858)
	at org.eclipse.esmf.aspectmodel.resolver.FileSystemStrategy$DefaultNamespace.apply(FileSystemStrategy.java:134)
	at org.eclipse.esmf.aspectmodel.resolver.FileSystemStrategy$DefaultNamespace.apply(FileSystemStrategy.java:116)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.getModelForUrn(AspectModelResolver.java:332)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.resolve(AspectModelResolver.java:262)
	at org.eclipse.esmf.aspectmodel.resolver.AspectModelResolver.resolveAspectModel(AspectModelResolver.java:182)
	... 27 more
```

--- a/core/esmf-aspect-meta-model-version-migrator/src/main/java/org/eclipse/esmf/aspectmodel/versionupdate/migrator/BammUriRewriter.java
+++ b/core/esmf-aspect-meta-model-version-migrator/src/main/java/org/eclipse/esmf/aspectmodel/versionupdate/migrator/BammUriRewriter.java
@@ -19,16 +19,15 @@ import java.util.Map;
 import java.util.Optional;
 import java.util.stream.Collectors;
 
-import org.eclipse.esmf.aspectmodel.resolver.exceptions.InvalidVersionException;
-import org.eclipse.esmf.aspectmodel.vocabulary.Namespace;
-import org.eclipse.esmf.samm.KnownVersion;
-
 import org.apache.jena.graph.NodeFactory;
 import org.apache.jena.rdf.model.Literal;
 import org.apache.jena.rdf.model.Model;
 import org.apache.jena.rdf.model.RDFNode;
 import org.apache.jena.rdf.model.ResourceFactory;
 import org.apache.jena.rdf.model.Statement;
+import org.eclipse.esmf.aspectmodel.resolver.exceptions.InvalidVersionException;
+import org.eclipse.esmf.aspectmodel.vocabulary.Namespace;
+import org.eclipse.esmf.samm.KnownVersion;
 
 /**
  * A {@link AbstractUriRewriter} that replaces all references to the legacy BAMM Aspect Meta Model to their corresponding SAMM counterparts
@@ -66,7 +65,7 @@ public class BammUriRewriter extends AbstractUriRewriter {
       // The mapping of the URNs of the legacy BAMM Aspect Meta model to their corresponding SAMM counterparts
       return Map.of(
             "urn:bamm:io.openmanufacturing:meta-model:" + bammVersion.versionString() + "#", targetPrefixes.get( "samm" ),
-            "urn:bamm:io.openmanufacturing:characteristic: " + bammVersion.versionString() + "#", targetPrefixes.get( SAMM_C_PREFIX ),
+            "urn:bamm:io.openmanufacturing:characteristic:" + bammVersion.versionString() + "#", targetPrefixes.get( SAMM_C_PREFIX ),
             "urn:bamm:io.openmanufacturing:entity:" + bammVersion.versionString() + "#", targetPrefixes.get( SAMM_E_PREFIX ),
             "urn:bamm:io.openmanufacturing:unit:" + bammVersion.versionString() + "#", targetPrefixes.get( "unit" )
       );

--- a/core/esmf-aspect-model-resolver/src/main/java/org/eclipse/esmf/aspectmodel/resolver/AspectModelResolver.java
+++ b/core/esmf-aspect-model-resolver/src/main/java/org/eclipse/esmf/aspectmodel/resolver/AspectModelResolver.java
@@ -13,8 +13,7 @@
 
 package org.eclipse.esmf.aspectmodel.resolver;
 
-import static io.vavr.API.$;
-import static io.vavr.API.Case;
+import static io.vavr.API.*;
 import static io.vavr.Predicates.instanceOf;
 
 import java.io.ByteArrayInputStream;
@@ -37,6 +36,13 @@ import java.util.function.Function;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
+import org.apache.commons.io.FilenameUtils;
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.rdf.model.Property;
+import org.apache.jena.rdf.model.RDFNode;
+import org.apache.jena.vocabulary.RDF;
+import org.apache.jena.vocabulary.XSD;
 import org.eclipse.esmf.aspectmodel.VersionNumber;
 import org.eclipse.esmf.aspectmodel.resolver.services.SammAspectMetaModelResourceResolver;
 import org.eclipse.esmf.aspectmodel.resolver.services.TurtleLoader;
@@ -51,17 +57,11 @@ import org.eclipse.esmf.aspectmodel.versionupdate.migrator.BammUriRewriter;
 import org.eclipse.esmf.samm.KnownVersion;
 
 import com.google.common.collect.Streams;
+
 import io.vavr.CheckedFunction1;
 import io.vavr.Value;
 import io.vavr.control.Option;
 import io.vavr.control.Try;
-import org.apache.commons.io.FilenameUtils;
-import org.apache.jena.rdf.model.Model;
-import org.apache.jena.rdf.model.ModelFactory;
-import org.apache.jena.rdf.model.Property;
-import org.apache.jena.rdf.model.RDFNode;
-import org.apache.jena.vocabulary.RDF;
-import org.apache.jena.vocabulary.XSD;
 
 /**
  * Provides facilities for loading an Aspect model and resolving referenced meta model elements and
@@ -330,10 +330,9 @@ public class AspectModelResolver {
             return Try.success( EMPTY_MODEL );
          }
          return resolutionStrategy.apply( aspectModelUrn ).flatMap( model -> {
-            if ( !model.contains( model.createResource( urn ), RDF.type, (RDFNode) null ) ) {
-               return Try.failure(
-                     new ModelResolutionException(
-                           "Resolution strategy returned a model which does not contain element definition for " + urn ) );
+            if ( !containsType( model, urn ) ) {
+               return Try.failure( new ModelResolutionException(
+                     "Resolution strategy returned a model which does not contain element definition for " + urn ) );
             }
             return Try.success( model );
          } );
@@ -342,6 +341,24 @@ public class AspectModelResolver {
          // to resolve, so we return just an empty model
          return Try.success( EMPTY_MODEL );
       }
+   }
+
+   private boolean containsType( final Model model, final String urn ) {
+      if ( model.contains( model.createResource( urn ), RDF.type, (RDFNode) null ) ) {
+         return true;
+      } else if ( urn.startsWith( "urn:samm:" ) ) {
+         // when deriving a URN from file (via "fileToUrn" method - mainly in samm-cli scenarios),
+         // we assume new "samm" format, but could actually have been the old "bamm"
+         return model.contains( model.createResource( toLegacyBammUrn( urn ) ), RDF.type, (RDFNode) null );
+      }
+      return false;
+   }
+
+   private String toLegacyBammUrn( final String urn ) {
+      if ( urn.startsWith( "urn:samm:" ) ) {
+         return urn.replace( "urn:samm:", "urn:bamm:" );
+      }
+      return urn;
    }
 
    /**

--- a/core/esmf-aspect-model-resolver/src/test/java/org/eclipse/esmf/aspectmodel/resolver/AspectModelResolverTest.java
+++ b/core/esmf-aspect-model-resolver/src/test/java/org/eclipse/esmf/aspectmodel/resolver/AspectModelResolverTest.java
@@ -15,31 +15,29 @@ package org.eclipse.esmf.aspectmodel.resolver;
 
 import static org.apache.jena.rdf.model.ResourceFactory.createResource;
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.junit.jupiter.api.Assertions.assertFalse;
-import static org.junit.jupiter.api.Assertions.assertTrue;
 
 import java.io.File;
 import java.net.URISyntaxException;
-import java.nio.file.Path;
+import java.nio.file.Paths;
 import java.util.List;
 
+import org.apache.jena.rdf.model.Model;
+import org.apache.jena.rdf.model.RDFNode;
+import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.rdf.model.Statement;
+import org.apache.jena.vocabulary.RDF;
 import org.eclipse.esmf.aspectmodel.resolver.services.TurtleLoader;
 import org.eclipse.esmf.aspectmodel.resolver.services.VersionedModel;
 import org.eclipse.esmf.aspectmodel.urn.AspectModelUrn;
 import org.eclipse.esmf.aspectmodel.vocabulary.SAMM;
 import org.eclipse.esmf.samm.KnownVersion;
 import org.eclipse.esmf.test.MetaModelVersions;
-
-import com.google.common.collect.Streams;
-import io.vavr.control.Try;
-import org.apache.jena.rdf.model.Model;
-import org.apache.jena.rdf.model.RDFNode;
-import org.apache.jena.rdf.model.Resource;
-import org.apache.jena.rdf.model.Statement;
-import org.apache.jena.vocabulary.RDF;
-import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
 import org.junit.jupiter.params.provider.MethodSource;
+
+import com.google.common.collect.Streams;
+
+import io.vavr.control.Try;
 
 public class AspectModelResolverTest extends MetaModelVersions {
    private final AspectModelResolver resolver = new AspectModelResolver();
@@ -75,6 +73,26 @@ public class AspectModelResolverTest extends MetaModelVersions {
                   .getPath() );
 
       final AspectModelUrn testUrn = AspectModelUrn.fromUrn( "urn:bamm:org.eclipse.esmf.test:2.0.0#BammTest" );
+
+      final ResolutionStrategy urnStrategy = new FileSystemStrategy( aspectModelsRootDirectory.toPath() );
+      final Try<VersionedModel> result = resolver.resolveAspectModel( urnStrategy, testUrn );
+      assertThat( result.isSuccess() ).isTrue();
+
+      final SAMM samm = new SAMM( KnownVersion.getLatest() );
+      assertThat( result.get().getModel().listStatements( null, RDF.type, samm.Aspect() ).nextOptional() ).isNotEmpty();
+   }
+
+   @ParameterizedTest
+   @MethodSource( value = "versionsUpToIncluding2_0_0" )
+   public void testLoadLegacyBammModelFromFileExpectSuccess( final KnownVersion metaModelVersion ) throws URISyntaxException {
+      final File aspectModelsRootDirectory = new File(
+            AspectModelResolverTest.class.getClassLoader()
+                  .getResource( metaModelVersion.toString().toLowerCase() )
+                  .toURI()
+                  .getPath() );
+
+      final AspectModelUrn testUrn = AspectModelResolver.fileToUrn(
+            Paths.get( aspectModelsRootDirectory.toString(), "org.eclipse.esmf.test", "2.0.0", "BammTest.ttl" ).toFile() );
 
       final ResolutionStrategy urnStrategy = new FileSystemStrategy( aspectModelsRootDirectory.toPath() );
       final Try<VersionedModel> result = resolver.resolveAspectModel( urnStrategy, testUrn );

